File: Python/Problems/Problems/problem1877.py
import heapq
from typing import List

from .problem import Problem


class Problem1877(Problem):
    def test(self) -> bool:
        nums = [3, 5, 2, 3]

        expected = 7

        result = self.minPairSum(nums)

        print(result)

        return expected == result

    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        half = len(nums) // 2
        n = len(nums)
        result = 0
        for i in range(half):
            result = max(result, nums[i] + nums[n - 1 - i])

        return result

File: C++/Problems/Problems/problem_1877.cpp
#include "problem_1877.h"

#include <algorithm>

bool problem_1877::test() {
    std::vector nums { 3, 5, 2, 3 };

    int expected = 7;

    auto result = minPairSum(nums);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_1877::minPairSum(std::vector<int> &nums) {
    std::ranges::sort(nums);
    int half = nums.size() / 2;
    int n = nums.size();
    int result = 0;
    for (int i = 0; i < half; ++i)
        result = std::max(result, nums[i] + nums[n - 1 - i]);

    return result;
}
